pkg/staticnetworkconfig/generator.go

```diff
@@ -148,6 +145,27 @@ func (s *StaticNetworkConfigGenerator) formatNMConnection(nmConnection string) (
 	return buf.String(), nil
 }
 
+func (s *StaticNetworkConfigGenerator) validateAndCalculateMAC(ifaceName string) (string, error) {
```

Comment:
this seems like  a good candidate for unit-test

Reply:
done